ConnectionFactory.php

```diff
@@ -52,12 +55,8 @@ public function createConnection(array $params, Configuration $config = null, Ev
                 $platform->registerDoctrineTypeMapping($dbType, $doctrineType);
             }
         }
-        if (! empty($this->commentedTypes)) {
-            $platform = $this->getDatabasePlatform($connection);
-            foreach ($this->commentedTypes as $type) {
-                $platform->markDoctrineTypeCommented(Type::getType($type));
-            }
-        }
+
+        $this->markTypesCommented($this->getDatabasePlatform($connection));
```

Comment:
We should avoid triggering the instantiation of the database platform if we have nothing to do (no custom types). This way, it avoids connecting to the DB early in case guessing the server version is needed to select the platform.

Reply:
btw, the Travis failure is exactly related to this, AFAICT.

Reply:
I've wrapped it and only mark types commented if the type config is non-empty.